lib/puppet/pops/lookup/interpolation.rb

```diff
@@ -90,14 +90,16 @@ def interpolate_method(method_key)
             lookup_invocation.report_found_in_overrides(root_key, ovr[root_key])
           else
             scope = lookup_invocation.scope
-            val = scope[root_key]
-            if val.nil? && !nil_in_scope?(scope, root_key)
```

Comment:
Thinking out loud, is it ok to not call the `nil_in_scope?` method? I think previously we called it to differentiate between the `root_key` had a `nil` value versus `root_key` is not assigned any value. So we needed something like `Scope#exists?` I think we may still need that check in the case where there isn't a default value (`lookup_invocation.default_values.include?(root_key)` returns false) and `Puppet[:strict]` is changed to off or warning?